### PR TITLE
[feat]: [CI-7491]: Add support for dot env and output vars for plugin

### PR DESCRIPTION
**Pull request by smjt-h**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46272)

### CODE REVIEW

File: product/ci/addon/tasks/plugin.go [hunk 1]
```diff
@@ -187,19 +190,31 @@ func (t *pluginTask) execute(ctx context.Context, retryCount int32) (map[string]
...
-	if len(t.envVarOutputs) != 0 && stepOutputExists {
+	_, err = t.fs.Stat(outputDotEnvFile)
+	if err == nil {
 		var err error
-		outputVars, err := fetchOutputVariables(outputFile, t.fs, t.log)
+		outputVars, err := fetchOutputVariablesFromEnv(outputDotEnvFile, t.log)
 		if err != nil {
-			logCommandExecErr(t.log, "error encountered while fetching output of the plugin step", t.id, "", retryCount, start, err)
+			logCommandExecErr(t.log, "error encountered while fetching output of the plugin step from .env File", t.id, "", retryCount, start, err)
 			return nil, nil, err
```
Comment (raghavharness) [inline product/ci/addon/tasks/plugin.go:200]: Let's not return error here for now if for some reason fetching output vars for plugin fails ?

File: product/ci/addon/tasks/plugin.go [hunk 2]
```diff
@@ -187,19 +190,31 @@ func (t *pluginTask) execute(ctx context.Context, retryCount int32) (map[string]
...
 		}
 
 		stepOutput = outputVars
+	} else {
```
Comment (raghavharness) [inline product/ci/addon/tasks/plugin.go:204]: Check once with @r0610205 if this is really needed ? Otherwise we can remove.

Comment (smjt-h) [reply product/ci/addon/tasks/plugin.go:204]: Yes it is needed as confirmed by @r0610205  so i have added in else block

File: product/ci/addon/tasks/plugin.go (hunk 1 above)
Comment (smjt-h) [reply product/ci/addon/tasks/plugin.go]: removed return from here

File: product/ci/addon/tasks/plugin.go (hunk 2 above)
Comment (shubham149) [reply product/ci/addon/tasks/plugin.go:204]: But output file is not passed anywhere. How will they use it?

File: product/ci/addon/tasks/common.go [hunk 3]
```diff
@@ -145,3 +147,23 @@ func fetchOutputVariables(outputFile string, fs filesystem.FileSystem, log *zap.
 	}
 	return envVarMap, nil
 }
+
+// Fetches map of env variable and value from .env output File.
+// OutputFile stores all env variable and value
+func fetchOutputVariablesFromEnv(outputFile string, log *zap.SugaredLogger) (
```
Comment (shubham149) [inline product/ci/addon/tasks/common.go:153]: fetchOutputVariablesFromDotEnv

**Comment by smjt-h** [comment]
trigger utAll

File: product/ci/addon/tasks/common.go (hunk 3 above)
Comment (smjt-h) [reply product/ci/addon/tasks/common.go]: done

**Review by shubham149 — APPROVED** [review]
> Reviewed 6 of 9 files at r1, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46272)* status: 2. unresolved discussions.

**Pull request merged**